src/models.ts

```diff
@@ -198,7 +198,7 @@ export abstract class Filter {
     this.target = target;
   }
 
-  toJSON(): IFilter {
+  toJSON(): any {
```

Comment:
Yes, then we should find an alternate way to specify the return type.  We should not be using `any`.